tv_apps/tv_shared/js/card_manager.js

```diff
@@ -206,51 +193,62 @@
       }
     },
 
+    _initCardStoreIfNeeded: function cm_initCardStore() {
+      if (!this._cardStore) {
+        this._cardStore =
+          new CardStore(this._mode, this._manifestURLOfCardStore);
+        this._cardStore.on('change', this._onCardStoreChange.bind(this));
+      }
+    },
+
+    // XXX: DO NOT call this directly except for _reloadCardList.
+    // Because it is not protected by AsyncSemaphore
+    _loadCardListFromCardStore:
```

Comment:
If we return a `Promise`, that means we are running this method in async. Then we'll have to protect it with AsyncSemaphore. I think it will be too complex for now. Maybe we could do this in follow up